addon/routes/application.js

```diff
@@ -2,6 +2,8 @@ import Route from "@ember/routing/route";
 import { inject as service } from "@ember/service";
 
 export default class ApplicationRoute extends Route {
+  queryParams = { activeGroup: { refreshModel: true, replace: false } };
```

Comment:
What is the reason for the `refreshModel`? As it seems your model is quite "static".

Reply:
I copy/pasted a solution from somewhere that worked. No questions asked :smile: 

I'll check if it works without the refresh. You're right that it probably will.

Reply:
I'll still test it. But thinking about it this will probably not work as the config update is inside the model hook. I'd say the hook is not rerun if the model is not "refreshed". I'll give you an update after the test.

Reply:
This is a missuse of the model hook, for this you should use `afterModel`. The refresh model is correct since you need the model to be rerun in order to do `this.config.activeGroup = params.activeGroup;` again.

Reply:
Yeah. The refresh is needed. But the replace was actually not a good choice. Changing the group should add to the browser history.

Reply:
I'll look into afterModel.

Reply:
@anehx How would you access the route and query params in the afterModel? Are they stored in the transition?

Reply:
Found them in the transition and moved the whole thing from the model to the afterModel hook and it seems to work :heart:

Reply:
Nice!